tests/multi_bluetooth/stress_log_filesystem.py

```diff
@@ -0,0 +1,185 @@
+        print("_IRQ_CENTRAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_CENTRAL_DISCONNECT:
+        print("_IRQ_CENTRAL_DISCONNECT")
+    elif event == _IRQ_PERIPHERAL_CONNECT:
+        print("_IRQ_PERIPHERAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_PERIPHERAL_DISCONNECT:
+        print("_IRQ_PERIPHERAL_DISCONNECT")
+    elif event == _IRQ_GATTC_SERVICE_RESULT:
+        # conn_handle, start_handle, end_handle, uuid = data
+        if data[-1] == SERVICE_UUID:
+            print("_IRQ_GATTC_SERVICE_RESULT", data[3])
+            waiting_events[event] = (data[1], data[2],)
+        else:
+            return
+    elif event == _IRQ_GATTC_SERVICE_DONE:
+        print("_IRQ_GATTC_SERVICE_DONE")
+    elif event == _IRQ_GATTC_CHARACTERISTIC_RESULT:
+        # conn_handle, def_handle, value_handle, properties, uuid = data
+        if data[-1] == CHAR_UUID:
+            print("_IRQ_GATTC_CHARACTERISTIC_RESULT", data[-1])
+            waiting_events[event] = data[2]
+        else:
+            return
+    elif event == _IRQ_GATTC_CHARACTERISTIC_DONE:
+        print("_IRQ_GATTC_CHARACTERISTIC_DONE")
+    elif event == _IRQ_GATTC_READ_RESULT:
+        print("_IRQ_GATTC_READ_RESULT", bytes(data[-1]))
+    elif event == _IRQ_GATTC_READ_DONE:
+        print("_IRQ_GATTC_READ_DONE", data[-1])
+    elif event == _IRQ_GATTC_WRITE_DONE:
+        print("_IRQ_GATTC_WRITE_DONE", data[-1])
+    elif event == _IRQ_GATTS_WRITE:
+        print("_IRQ_GATTS_WRITE")
+    elif event == _IRQ_GATTS_READ_REQUEST:
+        print("_IRQ_GATTS_READ_REQUEST")
+
+    if event not in waiting_events:
+        waiting_events[event] = None
+
+
+def wait_for_event(event, timeout_ms):
+    t0 = time.ticks_ms()
+    while time.ticks_diff(time.ticks_ms(), t0) < timeout_ms:
+        periodic_log_write()
+        if event in waiting_events:
+            result = waiting_events[event]
+            del waiting_events[event]
+            return result
+        machine.idle()
+    raise ValueError("Timeout waiting for {}".format(event))
+
+
+# Acting in peripheral role.
+def instance0():
+    global log_file
+    log_file = open(LOG_PATH_INSTANCE0, "w")
+    write_log('start')
+    ble.active(1)
+    ble.irq(irq)
+    multitest.globals(BDADDR=ble.config("mac"))
+    ((char_handle,),) = ble.gatts_register_services(SERVICES)
+    multitest.next()
+    try:
+        for repeat in range(2):
+            print("gap_advertise")
+            ble.gap_advertise(50_000, b"\x02\x01\x06\x04\xffMPY")
+            # Wait for central to connect, do a sequence of read/write, then disconnect.
+            wait_for_event(_IRQ_CENTRAL_CONNECT, TIMEOUT_MS)
+            for op in range(4):
+                wait_for_event(_IRQ_GATTS_READ_REQUEST, TIMEOUT_MS)
+                wait_for_event(_IRQ_GATTS_WRITE, TIMEOUT_MS)
+            wait_for_event(_IRQ_CENTRAL_DISCONNECT, 2*TIMEOUT_MS)
+    finally:
+        ble.active(0)
+        log_file.close()
```

Comment:
would it make sense for the test to re-open the file and print out its contents at the end here, for further verification it ran correctly?

Reply:
This file will not be exactly the same across runs.

Reply:
Ok